ext/oj/oj.h

```diff
@@ -379,6 +379,12 @@ extern bool oj_use_hash_alt;
 extern bool oj_use_array_alt;
 extern bool string_writer_optimized;
 
+static inline VALUE oj_safe_string_convert(VALUE obj) {
+    VALUE rstr = rb_funcall(obj, oj_to_s_id, 0);
```

Comment:
Should also verify the to_s method exists.

Reply:
```ruby
obj = BasicObject.new
p Oj.dump(obj, mode: :compat)
#=> json.rb:4:in `dump': undefined method `to_s' for #<BasicObject:0x0000559fe073d748> (NoMethodError)
```

`rb_funcall` raises a `NoMethodError` exception if `to_s` method does not exist.
I think this is expected behavior.

Reply:
True, it is good enough.